workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/utilities/TokenBuilder.kt

```diff
@@ -0,0 +1,113 @@
+package com.r3.corda.lib.tokens.workflows.utilities
+
+import com.r3.corda.lib.tokens.contracts.states.FungibleToken
+import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
+import com.r3.corda.lib.tokens.contracts.types.TokenType
+import com.r3.corda.lib.tokens.contracts.utilities.amount
+import com.r3.corda.lib.tokens.contracts.utilities.heldBy
+import com.r3.corda.lib.tokens.contracts.utilities.issuedBy
+import com.r3.corda.lib.tokens.contracts.utilities.of
+import com.r3.corda.lib.tokens.workflows.TokenBuilderException
+import net.corda.core.contracts.Amount
+import net.corda.core.identity.Party
+import java.lang.Exception
+import java.math.BigDecimal
+
+/**
+ * A utility class designed for Java developers to more easily access Kotlin DSL
+ * functions to build token types.
+ *
+ * This function vaguely follows a builder pattern, design choices were made
+ * to emulate Kotlin syntax as closely as possible for an easily transferable
+ * developer experience in Java.
+ */
+class TokenBuilder {
+    private var bigDecimalAmount: BigDecimal? = null
+    private lateinit var tokenType: TokenType
+    private lateinit var issuer: Party
+    private lateinit var holder: Party
+
+    /**
+     * Set the [bigDecimalAmount] member property of the builder using a provided [Long].
+     *
+     * @param longAmount The [Long] that will be converted to set the [bigDecimalAmount] member property
+     */
+    fun withAmountValue(longAmount: Long) = this.apply { bigDecimalAmount = BigDecimal.valueOf(longAmount) }
```

Comment:
`withAmount` might be simpler

Reply:
I thought this might be confusing given that `Amount` is already a class. In retrospect I think `withAmount` is still more clear. I'll revert to withAmount for now.